File: Code/BeforeLegends/Assets/Scripts/System/GameStateManager.js
//#pragma strict

//static var instance : GameStateManager;

var world : GameObject;
var battle : GameObject;

var state : int = 0;
var lastBattleResult : boolean = true;

//function Start(){
//	instance = this;
//}

//function startBattle(player : GameObject, enemy : GameObject){
//	battle.GetComponent(BattleController).init(player, enemy);
//	state = 1;
//	battle.SetActive(true);
//	world.SetActive(false);
//}

//function endBattle(result : boolean, exp : int){
//	state = 0;
//	world.SetActive(true);
//	battle.SetActive(false);
//	RessourceMaster.instance.valueScore += exp;
//	GameObject.Find("Olaf").GetComponent.<BattleParameters>().exp += exp;
//	GameObject.Find("Olaf").GetComponent.<BattleParameters>().LevelUp();
//}

File: Code/BeforeLegends/Assets/Scripts/System/GameStateManager.cs
using UnityEngine;
using System.Collections;

public class GameStateManager : MonoBehaviour {

    private static GameStateManager _instance;

    public static GameStateManager instance
    {
        get
        {
            if (_instance == null)
                _instance = GameObject.FindObjectOfType<GameStateManager>();
            return _instance;
        }
    }

    public GameObject world;
    public GameObject battle;
    public GameObject worldMapGUI;
    public GameObject battleMapGUI;
    public BattleParameters olaf;

    public int state;
    public bool lastBattleResult;

    public BattleController battleC;

    public void startBattle(GameObject player, GameObject enemy)
    {
		AudioMaster.instance.BattleStartA015Play();
        battleC.init(player, enemy);
        state = 1;
        battle.SetActive(true);
        world.SetActive(false);
        battleMapGUI.SetActive(true);
        worldMapGUI.SetActive(false);
    }

    public void endBattle(bool result, int exp)
    {
		AudioMaster.instance.audioSourceMusic.Stop();
	    state = 0;
	    world.SetActive(true);
        battle.SetActive(false);
        battleMapGUI.SetActive(false);
        worldMapGUI.SetActive(true);
	    ResourceManager.instance.ResourceAS("exp", exp);
	    olaf.exp += exp;
        olaf.LevelUp();
    }
}
